lib/src/value/calculation.dart

```diff
@@ -158,6 +343,204 @@ class SassCalculation extends Value {
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    _verifyCompatibleNumbers(args);
+    if ((dividend is! SassNumber || modulus is! SassNumber) ||
+        !dividend.hasCompatibleUnits(modulus)) {
+      return SassCalculation._("mod", args);
+    }
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    var strategyOrNumberSimplified = _simplify(strategyOrNumber);
+
+    switch ((
+      strategyOrNumberSimplified,
+      numberOrStep.andThen(_simplify),
+      step.andThen(_simplify)
+    )) {
+      case (SassNumber number, null, null):
+        return _matchUnits(number.value.round().toDouble(), number);
+
+      case (SassNumber number, SassNumber step, null):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep('nearest', number, step);
+
+      case (SassString strategy, SassNumber number, SassNumber step)
+          when {'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(strategy.text, number, step);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          SassNumber(),
+          null
+        ):
+        throw SassScriptException("If strategy is not null, step is required.");
+
+      case (SassString strategy, SassNumber number, null):
+        return SassCalculation._("round", [strategy, number]);
+
+      case (SassString(), _, _):
+      case (CalculationOperation(), _, null):
+      case (SassNumber(), _, null):
+        return SassCalculation._("round", [
+          strategyOrNumberSimplified,
+          if (numberOrStep != null) numberOrStep,
+          if (step != null) step
+        ]);
+
+      case (_, _, _):
+        throw SassScriptException(
+            "$strategyOrNumber must be either nearest, up, down or to-zero.");
```

Comment:
I think this catches more cases than just "strategy is set wrong", so it should probably be broken into separate cases with more specific error messages. For example, `round(up)` should complain about not having enough arguments, `round(1px + 10%)` should complain that a single-argument `round()` must be able to be simplified, and so on.

Reply:
Done, I added more tests to find any more edge cases and did some corrections. Thanks! :)